zlib/CMakeLists.txt

```
@@ -0,0 +1,18 @@
+set(COMPONENT_ADD_INCLUDEDIRS ./zlib)
+set(COMPONENT_SRCDIRS ./zlib)
+register_component()
+
+target_compile_options(${COMPONENT_LIB} PRIVATE  -Wno-unused-function)
+target_compile_options(${COMPONENT_LIB} PRIVATE  -Wno-implicit-int)
+target_compile_options(${COMPONENT_LIB} PRIVATE  -Wno-return-type)
+target_compile_options(${COMPONENT_LIB} PRIVATE  -Wno-format-extra-args)
+target_compile_options(${COMPONENT_LIB} PRIVATE  -Wno-unused-parameter)
+target_compile_options(${COMPONENT_LIB} PRIVATE  -Wno-unused-const-variable)
+target_compile_options(${COMPONENT_LIB} PRIVATE  -Wno-implicit-fallthrough)
+
+add_definitions(-DPPM_SUPPORTED)
```

Recommend using target-scoped CMake functions here:
```suggestion
target_compile_definitions(${COMPONENT_LIB} PRIVATE PPM_SUPPORTED)
```
(similar for the other lines)